### ISSUE TITLE
Probles in example_cmems.py 

### ISSUE THREAD

**Issue by ocecept** [OpenDrift/opendrift#150]
I tried to to run the example_cmems.py  and I got the following error:

File "../../examples/example_cmems.py", line 14, in <module>
    from opendrift.readers import reader_cmems
  File "/home/ocecept/opendrift/opendrift/readers/reader_cmems.py", line 28, in <module>
    from reader_netCDF_CF_generic import Reader as NCReader
ModuleNotFoundError: No module named 'reader_netCDF_CF_generic'

I than changed reader_cmems.py to:

from opendrift.readers.reader_netCDF_CF_generic import Reader as NCReader

And the model  runs and is able to produce the netcdf file with the CMEMS data, however it stops and  I get the error: 

INFO: "Missing variables: ['x_sea_water_velocity', 'y_sea_water_velocity']", 'The simulation stopped before requested end time was reached.'

I can see that the cmes output file has two variables uo and vo instead of x_sea_water_velocity and y_sea_water_velocity. I changed it using nco and run the model again recycling the files, but still no success. 

The log file is attached. 

Am I doing something wrong? 

Thanks
[LOG.TXT](https://github.com/OpenDrift/opendrift/files/3726257/LOG.TXT)

**Comment by gauteh** [comment]
Hi, this is likely caused by a typo in reader_cmems.py -- there is a small fix in #151. You could try that directly or wait a bit and we will merge it into master once we've tested it.

**Comment by gauteh** [comment]
Just read the rest of your comment properly! The log mentions that cmems reader is outside the time-coverage. Are you sure it provides data for the simulation time?

**Comment by ocecept** [comment]
Hi, thanks for your prompt replay. I think I understood the problem. Yes you are right, I did not get the time for all the simulation time. The script returns cmems data, but it did not cover all the time. I worked around it adding a "buffer" at time_end. 
 
I also found another problem when recycling data. 

 The simulation time is defined as "time = datetime.now()" and "time_end=time+duration", the script uses it to download data and to run the simulations. If I run the model, for example, at 8 am the scrip will run fine. However, if I try to recycle the data and run the simulations, lets say..  two hours later the model will not run, because time and time_end are different from the data that has been downloaded at 8 am.

**Comment by gauteh** [comment]
Yes, currently the CMEMS reader does not check if the cmems parameters changed, and it does not seem like the motu client does so either. We would either have to always re-download the file or store information about time and bounding box (possibly check if this is already in the netcdf file).
